src/scripts/add_targets_to_crawler/add_targets_to_crawler.py

```diff
@@ -0,0 +1,200 @@
+"""
+This module provides functions to update a Glue crawler by adding
+additional S3 targets.
+
+If a Glue crawler is configured to crawl new folders only, it cannot be updated
+via a regular stack update. Instead, we need to perform the following steps
+to completed the update:
+
+    1. Perform the desired crawler updates, but ensure that the update has a
+       recrawl policy of CRAWL_EVERYTHING so that the update completes successfully.
+    2. If the crawler originally had a different recrawl policy, we can
+       "undo" the change to the recrawl policy we made in the previous
+       step by doing an additional update -- but only to set the recrawl policy
+       back to its original value.
+
+This script supports a use case which occurs when we add support for
+new datasets. Additional S3 targets must be added to the crawlers, but our crawlers
+recrawl policy is to only crawl new folders. So the default behavior of the function
+`add_targets_to_crawler` is to add the S3 targets while overwriting the recrawl policy.
+We then update the crawler once more to change the recrawl policy back to crawl new
+folders only.
+"""
+import argparse
+import json
+import boto3
+
+CRAWL_EVERYTHING_POLICY = "CRAWL_EVERYTHING"
+CRAWLER_FORMAT = "{namespace}-{app}-{study}-{crawler_type}"
+TARGET_FORMAT = "s3://{bucket}/{namespace}/{app}/{study}/raw_json/dataset={dataset}/"
+
+def read_args():
+    parser = argparse.ArgumentParser(formatter_class=argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument(
+        "--environment",
+        required=True,
+        choices=["develop", "prod"],
+        help="The AWS environment (used to determine the S3 bucket name of each target)"
+    )
+    parser.add_argument(
+        "--crawler-type",
+        required=True,
+        choices=["standard", "array-of-records"],
+        help="The type of crawler"
+    )
+    parser.add_argument(
+        "--studies",
+        nargs="+",
+        help="List of study identifiers (e.g., cxhnxd)"
+    )
+    parser.add_argument(
+        "--datasets",
+        nargs="+",
+        help="List of dataset identifiers (e.g., ArchiveMetadata_v1)"
```

Comment:
I think you alluded a little to this, but what would it take to also validate that the studies and datasets are specified correctly?